clang/lib/Driver/ToolChains/ROCm.h

```diff
@@ -77,6 +79,82 @@ class RocmInstallationDetector {
           SPACKReleaseStr(SPACKReleaseStr.str()) {}
   };
 
+  struct CommonBitcodeLibsPreferences {
+    CommonBitcodeLibsPreferences(const Driver &D,
+                                 const llvm::opt::ArgList &DriverArgs,
+                                 StringRef GPUArch,
+                                 const Action::OffloadKind DeviceOffloadingKind,
+                                 const bool NeedsASanRT)
+        : ABIVer(DeviceLibABIVersion::fromCodeObjectVersion(
+              tools::getAMDGPUCodeObjectVersion(D, DriverArgs))) {
```

Comment:
Can we move the function definition to AMDGPU.cpp? It is pretty large.

Reply:
Sure, done.